src/Adapter/State/CommandHandler/EditStateHandler.php

```diff
@@ -0,0 +1,104 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\State\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\State\Command\EditStateCommand;
+use PrestaShop\PrestaShop\Core\Domain\State\CommandHandler\EditStateHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\CannotUpdateStateException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateException;
+use PrestaShop\PrestaShop\Core\Domain\State\Exception\StateNotFoundException;
+use PrestaShopException;
+use State;
+
+/**
+ * Handles state editing
+ */
+class EditStateHandler extends AbstractStateHandler implements EditStateHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws CannotUpdateStateException
+     * @throws StateConstraintException
+     * @throws StateException
+     * @throws StateNotFoundException
+     */
+    public function handle(EditStateCommand $command): void
+    {
+        $this->updateStateFromCommandData(
+            $this->getState($command->getStateId()),
+            $command
+        );
+    }
+
+    /**
+     * @param State $state
+     * @param EditStateCommand $command
+     *
+     * @throws CannotUpdateStateException
+     * @throws StateConstraintException
+     * @throws StateException
+     */
+    private function updateStateFromCommandData(State $state, EditStateCommand $command)
+    {
+        try {
+            if (null !== $command->getZoneId()->getValue()) {
+                $state->id_zone = $command->getZoneId()->getValue();
+            }
+
+            if (null !== $command->getCountryId()->getValue()) {
+                $state->id_country = $command->getCountryId()->getValue();
+            }
+
+            if (null !== $command->getIsoCode()) {
+                $state->iso_code = $command->getIsoCode();
+            }
+
+            if (null !== $command->getName()) {
+                $state->name = $command->getName();
+            }
+
+            if (null !== $command->getActive()) {
+                $state->active = $command->getActive();
+            }
+
```

Comment:
What would you suggest? How would you achieve this differently?

Reply:
No we use these `if` for partial update.

Reply:
My way to enhance code over iteration is to try to see and extract abstract pattern.

If i see some code like this, 

```php
if (null !== $command->getZoneId()->getValue()) {
    $state->id_zone = $command->getZoneId()->getValue();
}

if (null !== $command->getCountryId()->getValue()) {
    $state->id_country = $command->getCountryId()->getValue();
}

if (null !== $command->getIsoCode()) {
    $state->iso_code = $command->getIsoCode();
}

if (null !== $command->getName()) {
    $state->name = $command->getName();
}

if (null !== $command->getActive()) {
    $state->active = $command->getActive();
}
```

my first intention would be to reduce it this way :

```php
$state_setter_parameters = [
    'id_zone' => $command->getZoneId()->getValue(),
    'id_country' => $command->getCountryId()->getValue(),
    'iso_code' => $command->getIsoCode(),
    'name' => $command->getName(),
    'active' => $command->getActive(),
}

foreach($state_setter_parameters as $state_attribute => $state_attribute_value) {
    if (null !== $state_attribute_value) {
        $state->$state_attribute = $state_attribute_value;
    }

    // Or
    // $state->$state_attribute = $state_attribute_value ?? $state->$state_attribute;
}
```

Like this I dissociate logic and data and it would be easier to add some genericity later.

The key is to not go too far while you didn't define the needs. (In this example, don't consider adding default value strategy or different conditions to test while the need is defined)